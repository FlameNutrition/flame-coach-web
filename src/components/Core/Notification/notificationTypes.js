export const INFO = 'INFO';
export const WARNING = 'WARNING';
export const ERROR = 'ERROR';
export const SUCCESS = 'SUCCESS';
