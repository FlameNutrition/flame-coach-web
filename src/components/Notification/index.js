import React from 'react';
import {
  Box, Collapse, IconButton, makeStyles
} from '@material-ui/core';
import { Alert } from '@material-ui/lab';
import CloseIcon from '@material-ui/icons/Close';
import PropTypes from 'prop-types';
import clsx from 'clsx';

const useStyles = makeStyles(() => ({
  root: {},
  notificationBar: {
    marginTop: '10px'
  }
}));

const Notification = ({
  open,
  openHandler,
  message,
  level,
  collapse,
  className
}) => {
  const classes = useStyles();

  const NOTIFICATION_ERROR = 'error';
  const NOTIFICATION_WARNING = 'warning';
  const NOTIFICATION_INFO = 'info';
  const NOTIFICATION_SUCCESS = 'success';

  let internalLevel;

  switch (level) {
    case 'INFO':
      internalLevel = NOTIFICATION_INFO;
      break;
    case 'WARNING':
      internalLevel = NOTIFICATION_WARNING;
      break;
    case 'ERROR':
      internalLevel = NOTIFICATION_ERROR;
      break;
    case 'SUCCESS':
      internalLevel = NOTIFICATION_SUCCESS;
      break;
    default:
      internalLevel = NOTIFICATION_INFO;
      break;
  }

  return (
    <Box className={clsx(className, classes.notificationBar)}>
      {collapse
        ? (
          <Collapse in={open}>
            <Alert
              variant="filled"
              severity={internalLevel}
              action={(
                <IconButton
                  aria-label="close"
                  color="inherit"
                  size="small"
                  onClick={() => {
                    openHandler(false);
                  }}
                >
                  <CloseIcon fontSize="inherit" />
                </IconButton>
              )}
            >
              {message}
            </Alert>
          </Collapse>
        ) : (
          <Alert
            itemID="notification"
            variant="filled"
            severity={internalLevel}
          >
            {message}
          </Alert>
        )}
    </Box>
  );
};

Notification.propTypes = {
  message: PropTypes.string.isRequired,
  level: PropTypes.string.isRequired,
  open: PropTypes.bool,
  openHandler: PropTypes.func,
  collapse: PropTypes.bool,
  className: PropTypes.string
};

Notification.defaultProps = {
  collapse: false
};

export default Notification;
