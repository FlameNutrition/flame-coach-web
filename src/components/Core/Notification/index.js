import Box from "@material-ui/core/Box";
import Collapse from "@material-ui/core/Collapse";
import IconButton from "@material-ui/core/IconButton";
import { makeStyles } from "@material-ui/core";

import { ERROR, INFO, SUCCESS, WARNING } from "./notificationTypes";

import { Alert } from "@material-ui/lab";
import { Close as CloseIcon } from "@material-ui/icons";
import PropTypes from "prop-types";
import React from "react";
import clsx from "clsx";

const useStyles = makeStyles(() => ({
  root: {},
  notificationBar: {
    marginTop: "10px"
  }
}));

const Notification = ({
  open,
  openHandler,
  message,
  level,
  collapse,
  className
}) => {
  const classes = useStyles();

  let internalLevel;

  switch (level) {
    case INFO:
      internalLevel = "info";
      break;
    case WARNING:
      internalLevel = "warning";
      break;
    case ERROR:
      internalLevel = "error";
      break;
    case SUCCESS:
      internalLevel = "success";
      break;
    default:
      internalLevel = "info";
      break;
  }

  return (
    <Box className={clsx(className, classes.notificationBar)}>
      {collapse
        ? (
          <Collapse in={open}>
            <Alert
              variant="filled"
              severity={internalLevel}
              action={(
                <IconButton
                  aria-label="close"
                  color="inherit"
                  size="small"
                  onClick={() => {
                    openHandler(false);
                  }}
                >
                  <CloseIcon fontSize="inherit" />
                </IconButton>
              )}
            >
              {message}
            </Alert>
          </Collapse>
        ) : (
          <Alert
            itemID="notification"
            variant="filled"
            severity={internalLevel}
          >
            {message}
          </Alert>
        )}
    </Box>
  );
};

Notification.propTypes = {
  message: PropTypes.string.isRequired,
  level: PropTypes.string.isRequired,
  open: PropTypes.bool,
  openHandler: PropTypes.func,
  collapse: PropTypes.bool,
  className: PropTypes.string
};

Notification.defaultProps = {
  collapse: false
};

export default Notification;
