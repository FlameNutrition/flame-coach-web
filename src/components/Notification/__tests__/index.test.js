import React from 'react';
import sinon from 'sinon';
import { render, screen, cleanup, fireEvent } from '../../../testing/test-utils';
import Notification from '../index';

describe('<Notification/ >', () => {

  afterEach(() => {
    cleanup();
  });

  it('Show info notification with collapse', () => {
    const { container } = render(
      <Notification
        collapse message="INFO MESSAGE" level="INFO"/>
    );

    expect(container)
      .toMatchSnapshot();
    expect(screen.queryByText('INFO MESSAGE'))
      .not
      .toBeNull();
    expect(screen.queryByLabelText('close'))
      .not
      .toBeNull();
  });

  it('Message should be "I like you"', () => {
    render(
      <Notification message="I like you" level="SUCCESS"/>
    );

    expect(screen.queryByText('I like you'))
      .not
      .toBeNull();
  });

  it('Close notification', () => {
    const openHandlerMock = sinon.spy();
    render(
      <Notification
        collapse
        open
        message="INFO MESSAGE"
        level="SUCCESS"
        openHandler={openHandlerMock}
      />
    );

    fireEvent.click(screen.getByRole('button'));

    expect(openHandlerMock)
      .toHaveBeenCalledOnce(1);
  });
});
