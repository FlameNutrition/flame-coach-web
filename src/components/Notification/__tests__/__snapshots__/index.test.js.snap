// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification/ > Show info notification with collapse 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3 makeStyles-notificationBar-2"
  >
    <div
      class="MuiCollapse-root MuiCollapse-hidden"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            class="MuiPaper-root MuiAlert-root MuiAlert-filledInfo MuiPaper-elevation0"
            role="alert"
          >
            <div
              class="MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message"
            >
              INFO MESSAGE
            </div>
            <div
              class="MuiAlert-action"
            >
              <button
                aria-label="close"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
